File: SuperiorPackGroup/BLL/PhysicalInventoryWorksheetBLL.vb
Imports SuperiorPackGroup.SPGReportsTableAdapters

Public Class PhysicalInventoryWorksheetBLL

    Private m_PhysicalInventoryWorksheetTableAdapter As PhysicalInventoryWorksheetTableAdapter = Nothing

    Public ReadOnly Property Adapter() As PhysicalInventoryWorksheetTableAdapter
        Get
            If (Me.m_PhysicalInventoryWorksheetTableAdapter Is Nothing) Then
                Me.m_PhysicalInventoryWorksheetTableAdapter = New PhysicalInventoryWorksheetTableAdapter
                Me.m_PhysicalInventoryWorksheetTableAdapter.Connection.ConnectionString = My.Settings.UserConnectionString
            End If
            Return Me.m_PhysicalInventoryWorksheetTableAdapter
        End Get
    End Property

    Public Function GetPhysicalInventoryWorksheet(ByVal customerID As Nullable(Of Integer), ByVal itemType As String, ByVal items As String, ByVal inactiveItems As Boolean?, _
                                                  ByVal inactiveCustomers As Boolean?) As SPGReports.PhysicalInventoryWorksheetDataTable

        If inactiveCustomers.Value = True Then
            inactiveCustomers = Nothing
        End If

        If inactiveItems.Value = True Then
            inactiveItems = Nothing
        End If

        If String.IsNullOrEmpty(items) Then
            Return Adapter.GetPhysicalInventoryWorksheet(customerID, itemType, inactiveCustomers, inactiveItems)
        Else
            Return Adapter.GetPhysicalInventoryWorksheetByItemID(customerID, itemType, inactiveCustomers, items)
        End If

    End Function

End Class

File: SuperiorPackGroup/BLL/PhysicalInventoryWorksheetBLL.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using SuperiorPackGroup.SPGReportsTableAdapters;

namespace SuperiorPackGroup
{
	public class PhysicalInventoryWorksheetBLL
	{

		private PhysicalInventoryWorksheetTableAdapter m_PhysicalInventoryWorksheetTableAdapter = null;

		public PhysicalInventoryWorksheetTableAdapter Adapter
		{
			get
			{
				if (this.m_PhysicalInventoryWorksheetTableAdapter == null)
				{
					this.m_PhysicalInventoryWorksheetTableAdapter = new PhysicalInventoryWorksheetTableAdapter();
					this.m_PhysicalInventoryWorksheetTableAdapter.Connection.ConnectionString = Properties.Settings.Default.UserConnectionString;
				}
				return this.m_PhysicalInventoryWorksheetTableAdapter;
			}
		}

		public SPGReports.PhysicalInventoryWorksheetDataTable GetPhysicalInventoryWorksheet(int? customerID, string itemType, string items, bool? inactiveItems, bool? inactiveCustomers)
		{

			if (inactiveCustomers.Value == true)
			{
				inactiveCustomers = null;
			}

			if (inactiveItems.Value == true)
			{
				inactiveItems = null;
			}

			if (string.IsNullOrEmpty(items))
			{
				return Adapter.GetPhysicalInventoryWorksheet(customerID, itemType, inactiveCustomers, inactiveItems);
			}
			else
			{
				return Adapter.GetPhysicalInventoryWorksheetByItemID(customerID, itemType, inactiveCustomers, items);
			}

		}

	}

}